### PR TITLE
Memkind PMEM examples

### PR DESCRIPTION
**Pull request by abonisl**
Signed-off-by: Adrian Bonislawski <adrianx.bonislawski@intel.com>

---
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/80)

### CODE REVIEW

**Comment by michalbiesek** [comment]
Please extend commit message with description titles of example scenarios You introducing

**Comment by michalbiesek** [comment]
___
*[examples/Makefile.mk, line 2 at r1](https://reviewable.io/reviews/memkind/memkind/80#-LL_1u--1h_JjHqM75Am:-LL_1u--1h_JjHqM75An:b5jpv39) ([raw file](https://github.com/memkind/memkind/blob/573a0f4bafdd4a46453c3ca785ec34d0468ca285/examples/Makefile.mk#L2)):*
> ```
> 2016
> ```
2018

**Comment by michalbiesek** [comment]
___
*[examples/pmem_alignment.c, line 34 at r1](https://reviewable.io/reviews/memkind/memkind/80#-LL_2-6O-DjCZ_fenNsk:-LL_2-6O-DjCZ_fenNsl:bbdtxd1) ([raw file](https://github.com/memkind/memkind/blob/573a0f4bafdd4a46453c3ca785ec34d0468ca285/examples/pmem_alignment.c#L34)):*
> ```
> #include <memkind/internal/memkind_pmem.h>
> ```
remove internal API  - in whole PR

**Comment by michalbiesek** [comment]
___
*[examples/pmem_alignment.c, line 2 at r1](https://reviewable.io/reviews/memkind/memkind/80#-LL_2JEXE_zqOk2bSMYJ:-LL_2JEXE_zqOk2bSMYK:bxsk1bs) ([raw file](https://github.com/memkind/memkind/blob/573a0f4bafdd4a46453c3ca785ec34d0468ca285/examples/pmem_alignment.c#L2)):*
> ```
> 2015 - 2018
> ```
2018

**Comment by michalbiesek** [comment]
___
*[examples/pmem_kinds.c, line 2 at r1](https://reviewable.io/reviews/memkind/memkind/80#-LL_2Spr4eLytkqumYEf:-LL_2Sps3ENbWwPPPN97:bxsk1bs) ([raw file](https://github.com/memkind/memkind/blob/573a0f4bafdd4a46453c3ca785ec34d0468ca285/examples/pmem_kinds.c#L2)):*
> ```
> 2015 - 2018
> ```
2018

**Comment by michalbiesek** [comment]
___
*[examples/pmem_malloc.c, line 36 at r1](https://reviewable.io/reviews/memkind/memkind/80#-LL_2_zfDIoYH2RlbH8G:-LL_2_zfDIoYH2RlbH8H:b-3h0pv0) ([raw file](https://github.com/memkind/memkind/blob/573a0f4bafdd4a46453c3ca785ec34d0468ca285/examples/pmem_malloc.c#L36)):*
> ```
> include <sys/param.h>
> ```
remove all headers which are not needed.

**Comment by michalbiesek** [comment]
___
*[examples/pmem_malloc_unlimited.c, line 2 at r1](https://reviewable.io/reviews/memkind/memkind/80#-LL_2hlB3m2laBpb0DOi:-LL_2hlB3m2laBpb0DOj:bxsk1bs) ([raw file](https://github.com/memkind/memkind/blob/573a0f4bafdd4a46453c3ca785ec34d0468ca285/examples/pmem_malloc_unlimited.c#L2)):*
> ```
> 2015 - 2018
> ```
2018

**Comment by michalbiesek** [comment]
___
*[examples/pmem_malloc.c, line 46 at r1](https://reviewable.io/reviews/memkind/memkind/80#-LL_2mCH-SOHj_K2Ir_a:-LL_2mCH-SOHj_K2Ir_b:b-2s0cka) ([raw file](https://github.com/memkind/memkind/blob/573a0f4bafdd4a46453c3ca785ec34d0468ca285/examples/pmem_malloc.c#L46)):*
> ```C
> #include <unistd.h>
> 
> #define PMEM_MAX_SIZE	(MEMKIND_PMEM_MIN_SIZE * 2)
> ```

Remove Your change

**Comment by michalbiesek** [comment]
___
*[examples/pmem_multithreads.c, line 2 at r1](https://reviewable.io/reviews/memkind/memkind/80#-LL_2vH-9foBgHjgT6nB:-LL_2vH-9foBgHjgT6nC:bxsk1bs) ([raw file](https://github.com/memkind/memkind/blob/573a0f4bafdd4a46453c3ca785ec34d0468ca285/examples/pmem_multithreads.c#L2)):*
> ```
> 2015 - 2018
> ```
2018

**Comment by michalbiesek** [comment]
___
*[examples/pmem_usable_size.c, line 2 at r1](https://reviewable.io/reviews/memkind/memkind/80#-LL_2zMh1Uu2O4pBD8ns:-LL_2zMh1Uu2O4pBD8nt:bxsk1bs) ([raw file](https://github.com/memkind/memkind/blob/573a0f4bafdd4a46453c3ca785ec34d0468ca285/examples/pmem_usable_size.c#L2)):*
> ```
> 2015 - 2018
> ```
2018

**Comment by michalbiesek** [comment]
___
*[examples/pmem_malloc.c, line 36 at r1](https://reviewable.io/reviews/memkind/memkind/80#-LL_2_zfDIoYH2RlbH8G:-LL_30YD2Tt9kuXK9nwI:bv0spfs) ([raw file](https://github.com/memkind/memkind/blob/573a0f4bafdd4a46453c3ca785ec34d0468ca285/examples/pmem_malloc.c#L36)):*
<details><summary><i>Previously, michalbiesek wrote…</i></summary><blockquote>

> ```
> include <sys/param.h>
> ```
remove all headers which are not needed.
</blockquote></details>

This concern whole PR

**Comment by michalbiesek** [comment]
___
*[examples/README.md, line 55 at r1](https://reviewable.io/reviews/memkind/memkind/80#-LL_4haJ6EFBEBIy5qM0:-LL_4haJ6EFBEBIy5qM1:b-jhgvhz) ([raw file](https://github.com/memkind/memkind/blob/573a0f4bafdd4a46453c3ca785ec34d0468ca285/examples/README.md#L55)):*
> ```
> usefull
> ```
useful

**Comment by michalbiesek** [comment]
___
*[examples/README.md, line 51 at r1](https://reviewable.io/reviews/memkind/memkind/80#-LL_4uxuEZE0PYOS6jNm:-LL_4uxv2PAKzuSkHgjD:b-hxnx7d) ([raw file](https://github.com/memkind/memkind/blob/573a0f4bafdd4a46453c3ca785ec34d0468ca285/examples/README.md#L51)):*
> ```
> aligment
> ```
alignment

**Comment by michalbiesek** [comment]
Update with new file references:
https://github.com/memkind/memkind/blob/54c7ab245ad357e732755f25a0fe8a63ae85c61f/MANIFEST#L29
https://github.com/memkind/memkind/blob/54c7ab245ad357e732755f25a0fe8a63ae85c61f/copying_headers/MANIFEST.freeBSD3#L1

**Comment by michalbiesek** [comment]
.
https://github.com/memkind/memkind/blob/54c7ab245ad357e732755f25a0fe8a63ae85c61f/test/Makefile.mk#L235

**Comment by michalbiesek** [comment]
___
*[README, line 50 at r2](https://reviewable.io/reviews/memkind/memkind/80#-LLiT-O-Ay29DF5wjkue:-LLiT-O-Ay29DF5wjkuf:b-r59py9) ([raw file](https://github.com/memkind/memkind/blob/d7386572ce1a467a904698e0b0560135c8f9db0f/README#L50)):*
> ```
>     man hbwmalloc
> 
> BUILDING
> ```

remove this change - do it in another PR - this isn't related to exapmles

**Comment by michalbiesek** [comment]
___
*[examples/pmem_alignment.c, line 139 at r2](https://reviewable.io/reviews/memkind/memkind/80#-LLiTGLX9vodvFF_CI1D:-LLiTGLY3ZhanfTJN8Ky:bdi5od5) ([raw file](https://github.com/memkind/memkind/blob/d7386572ce1a467a904698e0b0560135c8f9db0f/examples/pmem_alignment.c#L139)):*
> ```C
>     }
> 
>     printf("The memory has been successfully allocated using memkind alignment.\n");
> ```

fprintf(stdout

**Comment by michalbiesek** [comment]
___
*[examples/pmem_malloc.c, line 46 at r1](https://reviewable.io/reviews/memkind/memkind/80#-LL_2mCH-SOHj_K2Ir_a:-LLiTnW229lExi9wE3zL:bdjyxp) ([raw file](https://github.com/memkind/memkind/blob/573a0f4bafdd4a46453c3ca785ec34d0468ca285/examples/pmem_malloc.c#L46)):*
<details><summary><i>Previously, abonisl (abonisl) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Please use #define PMEM_MAX_SIZE (MEMKIND_PMEM_MIN_SIZE * 2)

**Comment by michalbiesek** [comment]
___
*[examples/pmem_kinds.c, line 124 at r2](https://reviewable.io/reviews/memkind/memkind/80#-LLiUNDmESgW0qPMxXnV:-LLiUNDmESgW0qPMxXnW:b-5kx5ux) ([raw file](https://github.com/memkind/memkind/blob/d7386572ce1a467a904698e0b0560135c8f9db0f/examples/pmem_kinds.c#L124)):*
> ```
> has
> ```
have

**Comment by michalbiesek** [comment]
___
*[examples/pmem_kinds.c, line 63 at r2](https://reviewable.io/reviews/memkind/memkind/80#-LLiWpXG9Z6ytyURkogH:-LLiWpXG9Z6ytyURkogI:b-gvbi9p) ([raw file](https://github.com/memkind/memkind/blob/d7386572ce1a467a904698e0b0560135c8f9db0f/examples/pmem_kinds.c#L63)):*
<details><summary><i alt="summary">Quoted 17 lines of code…</i></summary>

> ```
> while((opt = getopt(argc, argv, "hd:")) != -1) {
>         switch (opt) {
>             case 'd':
>                 PMEM_DIR = optarg;
>                 err = stat(PMEM_DIR, &st);
>                  if (err != 0 || !S_ISDIR(st.st_mode)) {
>                     printf("%s : Error in getting path status or"
>                            "invalid or non-existent directory\n", PMEM_DIR);
>                     return -1;
>                 }
>                  break;
>             case 'h':
>                 printf("\nMemkind options:\n"
>                        "-d <directory_path>   change directory on which PMEM kinds\n"
>                        "                      are created (default /tmp/)\n");
>                 break;
>         }
> ```
</details>

Not needed - You don't need to use help parameter.
Do better something like:
	if (argc > 2)
	{
			fprintf(stderr,"usage: %s [pmem_kind_dir_path], argv[0]);
			return 1;
	}
	if (argc == 2)
	{
	    if (stat(argv[1], &st) != 0 || !S_ISDIR(st.st_mode)) {
                    fprintf(stderr,"%s : Invalid path to pmem kind directory " argv[1]);
                    return 1;
		}
	}

**Comment by michalbiesek** [comment]
___
*[examples/pmem_kinds.c, line 44 at r2](https://reviewable.io/reviews/memkind/memkind/80#-LLiYXHVAfPEOCecIV-d:-LLiYXHVAfPEOCecIV-e:b-khjy31) ([raw file](https://github.com/memkind/memkind/blob/d7386572ce1a467a904698e0b0560135c8f9db0f/examples/pmem_kinds.c#L44)):*
> ```C
> #define PMEM_MAX_SIZE (1024 * 1024 * 32)
> 
> char* PMEM_DIR = "/tmp/";
> ```

static

**Comment by michalbiesek** [comment]
___
*[examples/README.md, line 3 at r2](https://reviewable.io/reviews/memkind/memkind/80#-LLiYayT6g1J9ZSHsXWt:-LLiYayT6g1J9ZSHsXWu:b4boz3h) ([raw file](https://github.com/memkind/memkind/blob/d7386572ce1a467a904698e0b0560135c8f9db0f/examples/README.md#L3)):*
> ```Markdown
> # Memkind examples
> 
> The example directory contains some example codes that use the memkind
> ```

Please provide information that it is possible to execute examples with user defined directory (where create pmem) and if it is not defined information what is default path

**Comment by michalbiesek** [comment]
___
*[examples/pmem_multithreads.c, line 44 at r2](https://reviewable.io/reviews/memkind/memkind/80#-LLiZIwYC2rL_gqRvZiR:-LLiZIwYC2rL_gqRvZiS:b-ep67pv) ([raw file](https://github.com/memkind/memkind/blob/d7386572ce1a467a904698e0b0560135c8f9db0f/examples/pmem_multithreads.c#L44)):*
> ```
> PMEM_MAX_SIZE
> ```
replace this with definition like (2*MEMKIND_PMEM_MIN_SIZE)
consider rename this variable for PMEM_SIZE.

**Comment by michalbiesek** [comment]
___
*[examples/pmem_alignment.c, line 37 at r2](https://reviewable.io/reviews/memkind/memkind/80#-LLioLME7Qto2NpyBolA:-LLioLME7Qto2NpyBolB:bh5hib3) ([raw file](https://github.com/memkind/memkind/blob/d7386572ce1a467a904698e0b0560135c8f9db0f/examples/pmem_alignment.c#L37)):*
> ```C
> #include <stdio.h>
> #include <errno.h>
> #include <ctype.h>
> ```

#include <ctype.h>
#include <stdlib.h>
#include <unistd.h>
I think You don't need to include this libraries.
This concern other examples.

**Comment by michalbiesek** [comment]
___
*[test/Makefile.mk, line 2 at r3](https://reviewable.io/reviews/memkind/memkind/80#-LLixJvk8wpOwi-Bqj61:-LLixJvlBP9WDCzv4jLO:b7okjyc) ([raw file](https://github.com/memkind/memkind/blob/2544d8b936f707d81cb2580384763db42fc3fc86/test/Makefile.mk#L2)):*
> ```
> 2017
> ```
2018

**Comment by michalbiesek** [comment]
___
*[examples/pmem_kinds.c, line 45 at r3](https://reviewable.io/reviews/memkind/memkind/80#-LLj3woXBkXEe4kxv0u5:-LLj3woY63376cHDH8i7:b1a80y1) ([raw file](https://github.com/memkind/memkind/blob/2544d8b936f707d81cb2580384763db42fc3fc86/examples/pmem_kinds.c#L45)):*
> ```
> *pmem_kind = NULL;
> *pmem_kind_unlimited = NULL;
> ```
Please do it as:
struct memkind *pmem_kind = NULL;
struct memkind *pmem_kind_unlimited = NULL;

**Comment by michalbiesek** [comment]
___
*[examples/pmem_malloc.c, line 2 at r3](https://reviewable.io/reviews/memkind/memkind/80#-LLj5D0M0adm2R7Ypelj:-LLj5D0M0adm2R7Ypelk:bz6yyun) ([raw file](https://github.com/memkind/memkind/blob/2544d8b936f707d81cb2580384763db42fc3fc86/examples/pmem_malloc.c#L2)):*
> ```C
> /*
>  * Copyright (c) 2018 Intel Corporation
> ```

2015 - 2018

This file exist before Your changes - You moved it ( rename it )

**Comment by michalbiesek** [comment]
___
*[examples/pmem_malloc.c, line 111 at r3](https://reviewable.io/reviews/memkind/memkind/80#-LLj5Nv99khpRT6lO6GL:-LLj5Nv99khpRT6lO6GM:bvag2pn) ([raw file](https://github.com/memkind/memkind/blob/2544d8b936f707d81cb2580384763db42fc3fc86/examples/pmem_malloc.c#L111)):*
> ```C
>     }
> 
>     sprintf(pmem_str10, "Hello world from persistent memory!\n");
> ```

Remove this change

**Comment by michalbiesek** [comment]
___
*[examples/README.md, line 3 at r3](https://reviewable.io/reviews/memkind/memkind/80#-LLj7LOG0UAUtxoe7EQk:-LLj7LOG0UAUtxoe7EQl:bcl9vax) ([raw file](https://github.com/memkind/memkind/blob/2544d8b936f707d81cb2580384763db42fc3fc86/examples/README.md#L3)):*
> ```
> some
> ```
remove "some"

**Comment by michalbiesek** [comment]
___
*[examples/pmem_usable_size.c, line 62 at r3](https://reviewable.io/reviews/memkind/memkind/80#-LLj65LzEnT4kYCeLXgN:-LLj65LzEnT4kYCeLXgO:b29v1yy) ([raw file](https://github.com/memkind/memkind/blob/2544d8b936f707d81cb2580384763db42fc3fc86/examples/pmem_usable_size.c#L62)):*
> ```
> An example showing
> ```
Cosmetics, but please consist with beginning explanation what test do. 
Currently You have:
- An example showing
-Simple example how 
-This example shows

Choose one form ("This example shows")and use it in all examples.

**Comment by michalbiesek** [comment]
___
*[examples/pmem_alignment.c, line 91 at r3](https://reviewable.io/reviews/memkind/memkind/80#-LLj9AQzEHe0ATYMcI7W:-LLj9AQzEHe0ATYMcI7X:b-6nv3hj) ([raw file](https://github.com/memkind/memkind/blob/2544d8b936f707d81cb2580384763db42fc3fc86/examples/pmem_alignment.c#L91)):*
> ```
> Probably
> ```
remove "Probably"

**Comment by michalbiesek** [comment]
___
*[examples/README.md, line 9 at r3](https://reviewable.io/reviews/memkind/memkind/80#-LLjBenz5tocnxYgJRYX:-LLjBenz5tocnxYgJRYY:b-fbjm2e) ([raw file](https://github.com/memkind/memkind/blob/2544d8b936f707d81cb2580384763db42fc3fc86/examples/README.md#L9)):*
> ```Markdown
> 
> The pmem_*.c demonstrates how to create and use a file-backed memkind.
>  The default pmem path is "/tmp/", to provide custom directory please use:
> ```

Remove space at the begining.

**Comment by michalbiesek** [comment]
___
*[examples/README.md, line 8 at r3](https://reviewable.io/reviews/memkind/memkind/80#-LLjDB3FDdLcxVaHi4Ph:-LLjDB3G2gqvVeX13Eoa:b1x0ypt) ([raw file](https://github.com/memkind/memkind/blob/2544d8b936f707d81cb2580384763db42fc3fc86/examples/README.md#L8)):*
> ```
> file-backed memkind
> ```
file-backed memory kind

**Comment by michalbiesek** [comment]
___
*[examples/README.md, line 11 at r3](https://reviewable.io/reviews/memkind/memkind/80#-LLjDV0LDWsgCfEgU_uX:-LLjDV0LDWsgCfEgU_uY:b-6pswep) ([raw file](https://github.com/memkind/memkind/blob/2544d8b936f707d81cb2580384763db42fc3fc86/examples/README.md#L11)):*
> ```
>to provide custom directory please use:
>  ./pmem_malloc.c /custom/pmem/dir/
> ```
You could rephrase it to something like:

Custom directory is pass as first argument to all of PMEM example programs, e.g. to execute pmem_malloc example in /mnt/pmem location, call:
./pmem_malloc /mnt/pmem/

**Comment by michalbiesek** [comment]
.

https://github.com/memkind/memkind/blob/54c7ab245ad357e732755f25a0fe8a63ae85c61f/copying_headers/MANIFEST.freeBSD3#L1

**Comment by michalbiesek** [comment]
___
*[examples/pmem_malloc_unlimited.c, line 48 at r3](https://reviewable.io/reviews/memkind/memkind/80#-LLjFFEn2cTqjOfSzDji:-LLmmzUW258vusbhRN2a:bhz134b) ([raw file](https://github.com/memkind/memkind/blob/2544d8b936f707d81cb2580384763db42fc3fc86/examples/pmem_malloc_unlimited.c#L48)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

> ```
>  
>     if (argc > 2) {
>         fprintf(stderr,"Usage: %s [pmem_kind_dir_path]", argv[0]);
>         return 1;
>     }
>     if (argc == 2) {
>         if (stat(argv[1], &st) != 0 || !S_ISDIR(st.st_mode)) {
>             fprintf(stderr,"%s : Invalid path to pmem kind directory ", argv[1]);
>             return 1;
>         } else
>             PMEM_DIR = argv[1];
>     }
>  
>     fprintf(stdout,
>             "Memory allocation example with the use of unlimited kind size.\nPMEM kind directory: %s\n",
>             PMEM_DIR);
> ```

I see this code block few times without any change. It's a good practice to delegate it to another file.
</blockquote></details>

I know this is basicly copy&paste, but assumed scenario You are installed memkind. In current situation You could basicly copy&paste code from example to Your new file link with lmemkind and Your new file will compile. I don't want this in seperate header for example - maybe delegate this argument parsing to function will clear the example code - but I don't want it in separate file.

**Comment by michalbiesek** [comment]
___
*[examples/pmem_malloc.c, line 111 at r3](https://reviewable.io/reviews/memkind/memkind/80#-LLj5Nv99khpRT6lO6GL:-LM20PTWE7JbFHIERtxj:b-828cww) ([raw file](https://github.com/memkind/memkind/blob/2544d8b936f707d81cb2580384763db42fc3fc86/examples/pmem_malloc.c#L111)):*
<details><summary><i>Previously, abonisl (abonisl) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

By Remove this change I mean only Your change :)
Please restore:
sprintf(pmem_str10, "Hello world from persistent memory\n");
fprintf(stdout, "%s", pmem_str10);

**Comment by michalbiesek** [comment]
___
*[examples/pmem_multithreads.c, line 89 at r5](https://reviewable.io/reviews/memkind/memkind/80#-LM261KF5dQx4h86flGg:-LM261KF5dQx4h86flGh:b111nbm) ([raw file](https://github.com/memkind/memkind/blob/84d2554ed8b6ca8fed1d35fa427bd982904dc0ed/examples/pmem_multithreads.c#L89)):*
> ```
> int *pmem_tint[10][100], t, i;
> ```
separate this 
 int *pmem_tint[10][100];
int t, i;

**Comment by michalbiesek** [comment]
___
*[examples/pmem_multithreads.c, line 88 at r5](https://reviewable.io/reviews/memkind/memkind/80#-LM268W41wcKp5Gp2j6h:-LM268W41wcKp5Gp2j6i:bytadog) ([raw file](https://github.com/memkind/memkind/blob/84d2554ed8b6ca8fed1d35fa427bd982904dc0ed/examples/pmem_multithreads.c#L88)):*
> ```
> 10
> ```
#define NUM_THREADS 10 at the beginning of the file

**Comment by michalbiesek** [comment]
___
*[examples/pmem_multithreads.c, line 88 at r5](https://reviewable.io/reviews/memkind/memkind/80#-LM268W41wcKp5Gp2j6h:-LM26Ubp67vdwe7_BS9U:b-64zyoy) ([raw file](https://github.com/memkind/memkind/blob/84d2554ed8b6ca8fed1d35fa427bd982904dc0ed/examples/pmem_multithreads.c#L88)):*
<details><summary><i>Previously, michalbiesek wrote…</i></summary><blockquote>

> ```
> 10
> ```
#define NUM_THREADS 10 at the beginning of the file
</blockquote></details>

And replace occurence of magic number ("10")

**Comment by michalbiesek** [comment]
___
*[examples/pmem_multithreads.c, line 89 at r6](https://reviewable.io/reviews/memkind/memkind/80#-LM2GZWP2lLhgyGmLjF2:-LM2GZWP2lLhgyGmLjF3:b-t00wlw) ([raw file](https://github.com/memkind/memkind/blob/6cac9bead59f8ca0694934e61253ab657165e602/examples/pmem_multithreads.c#L89)):*
> ```
> 10
> ```
NUM_THREADS

**Comment by michalbiesek** [comment]
___
*[examples/README.md, line 36 at r6](https://reviewable.io/reviews/memkind/memkind/80#-LMBw0XpA1Axn6lh1vl3:-LMBw0XpA1Axn6lh1vl4:bkpbr6n) ([raw file](https://github.com/memkind/memkind/blob/6cac9bead59f8ca0694934e61253ab657165e602/examples/README.md#L36)):*
> ```Markdown
> ### pmem_multithreads.c
> 
> This example shows how to use multithreading with pmem kinds.
> ```

As we spoke please extend description.

**Comment by michalbiesek** [comment]
Following places should be updated:

https://github.com/memkind/memkind/blob/04a8bc43ac7313918fdba0ecbd9bc79aa1667c61/.gitignore#L92 - please update references to Your examples ( remove the existing one )
https://github.com/memkind/memkind/blob/04a8bc43ac7313918fdba0ecbd9bc79aa1667c61/.gitignore#L73 - please update references to Your examples which will be generated in test ( remove existing one )
https://github.com/memkind/memkind/blob/04a8bc43ac7313918fdba0ecbd9bc79aa1667c61/memkind.spec.mk#L197  - please update references to Your examples ( remove the existing one )

**Comment by abonisl** [comment]
All done, sorry for commit mess. Waiting for final review
